src/adapter/src/coord/ddl.rs

```diff
@@ -476,6 +476,44 @@ impl Coordinator {
         Ok(result)
     }
 
+    async fn drop_replica(&mut self, cluster_id: ClusterId, replica_id: ReplicaId) {
+        if let Some(Some(ReplicaMetadata {
+            last_heartbeat,
+            metrics,
+            write_frontiers,
+        })) = self.transient_replica_metadata.insert(replica_id, None)
+        {
+            let mut updates = vec![];
+            if let Some(last_heartbeat) = last_heartbeat {
+                let retraction = self.catalog.state().pack_replica_heartbeat_update(
+                    replica_id,
+                    last_heartbeat,
+                    -1,
+                );
+                updates.push(retraction);
+            }
+            if let Some(metrics) = metrics {
+                let retraction = self
+                    .catalog
+                    .state()
+                    .pack_replica_metric_updates(replica_id, &metrics, -1);
+                updates.extend(retraction.into_iter());
+            }
+            let retraction = self.catalog.state().pack_replica_write_frontiers_updates(
+                replica_id,
+                &write_frontiers,
+                -1,
+            );
+            updates.extend(retraction.into_iter());
+            self.send_builtin_table_updates(updates, BuiltinTableUpdateSource::Background)
+                .await;
+        }
+        self.controller
+            .drop_replica(cluster_id, replica_id)
+            .await
+            .expect("dropping replica must not fail");
+    }
```

Comment:
This is pure code movement out of `sequencer.rs`. It's only used it this file and logically belongs in this file.